cmd/repository/default-pkg-registry.go

```diff
@@ -22,89 +24,55 @@ Further improvements could store registry as indexes, and laze load
 further information when necessary to reduce the startup time
 */
 type defaultRegistry struct {
-	packages       map[string]defaultRegistryEntry
-	groupCmds      map[string]*command.DefaultCommand // key is in form of [group]_[cmd name] ex. "_hotfix"
-	executableCmds map[string]*command.DefaultCommand // key is in form of [group]_[cmd name] ex. "hotfix_create"
-}
-
-type defaultRegistryEntry struct {
-	PkgName     string                    `json:"pkgName"`
-	PkgVersion  string                    `json:"version"`
-	PkgCommands []*command.DefaultCommand `json:"cmds"`
-}
-
-func (pkg *defaultRegistryEntry) Name() string {
-	return pkg.PkgName
-}
-
-func (pkg *defaultRegistryEntry) Version() string {
-	return pkg.PkgVersion
+	packages       map[string]command.PackageManifest
+	groupCmds      map[string]command.Command // key is in form of [group]_[cmd name] ex. "_hotfix"
+	executableCmds map[string]command.Command // key is in form of [group]_[cmd name] ex. "hotfix_create"
 }
 
-func (pkg *defaultRegistryEntry) Commands() []command.Command {
-	cmds := []command.Command{}
-	for _, c := range pkg.PkgCommands {
-		cmds = append(cmds, c)
-	}
-	return cmds
-}
-
-func NewRegistryEntry(pkg command.Package, pkgDir string) defaultRegistryEntry {
-	defPkg := defaultRegistryEntry{
-		PkgName:     pkg.Name(),
-		PkgVersion:  pkg.Version(),
-		PkgCommands: []*command.DefaultCommand{},
+func newDefaultRegistry() (Registry, error) {
+	reg := defaultRegistry{
+		packages:       make(map[string]command.PackageManifest),
+		groupCmds:      make(map[string]command.Command),
+		executableCmds: make(map[string]command.Command),
 	}
 
-	for _, cmd := range pkg.Commands() {
-		newCmd := command.NewDefaultCommandFromCopy(cmd, pkgDir)
-		defPkg.PkgCommands = append(defPkg.PkgCommands, newCmd)
-	}
-
-	return defPkg
+	return &reg, nil
 }
 
-func LoadRegistry(pathname string) (*defaultRegistry, error) {
-	registry := defaultRegistry{
-		packages:       make(map[string]defaultRegistryEntry),
-		groupCmds:      make(map[string]*command.DefaultCommand),
-		executableCmds: make(map[string]*command.DefaultCommand),
-	}
-
-	_, err := os.Stat(pathname)
+func (reg *defaultRegistry) Load(repoDir string) error {
+	_, err := os.Stat(repoDir)
 	if !os.IsNotExist(err) {
-		payload, err := ioutil.ReadFile(pathname)
+		files, err := os.ReadDir(repoDir)
 		if err != nil {
-			return nil, err
+			log.Errorf("cannot read the repo dir: %v", err)
+			return err
 		}
 
-		err = json.Unmarshal(payload, &registry.packages)
-		if err != nil {
-			return nil, err
+		for _, f := range files {
+			if !f.IsDir() && f.Type()&os.ModeSymlink != os.ModeSymlink {
+				continue
+			}
+			if dropinPkgManifestFile, err := os.Open(filepath.Join(repoDir, f.Name(), "manifest.mf")); err == nil {
```

Comment:
we shouldn't name it dropinPkgManifestFile here. In this particular case, it is a general case, not only for dropin